### PR TITLE
Replace TAO checks with same origin check

### LINKED ISSUE #104: "timing allow check" is not well defined in the context of NavTiming
https://w3c.github.io/navigation-timing/#dom-performancenavigationtiming-unloadeventstart says:

> If the previous document, and all HTTP redirects when navigating, pass the timing allow check algorithm...

But "timing allow check" as defined at https://w3c.github.io/resource-timing/#dfn-timing-allow-check is something that operates on "resources" (which are not really a defined concept, as far as I can tell).  In any case, it assumes that a "resource" has a Request attached to it, which a document may well not have at all.  

It's really not clear to me what behavior is supposed to be implemented here and why it doesn't lead to security bugs depending on what people decide to do with Timing-Allow-Origin headers in this nonsensical case...

### PR DESCRIPTION
**Pull request by yoavweiss**
Fixes #104

***
<a href="https://pr-preview.s3.amazonaws.com/yoavweiss/navigation-timing/pull/108.html" title="Last updated on Jun 19, 2019, 10:28 PM UTC (2ee06a5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/navigation-timing/108/396803f...yoavweiss:2ee06a5.html" title="Last updated on Jun 19, 2019, 10:28 PM UTC (2ee06a5)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -938,9 +925,8 @@ <h3>
           instance, the unloading time reveals how long the previous page takes
           to execute its unload handler, which could be used to infer the
           user's login status. These attacks have been mitigated by enforcing
-          the <a data-cite="resource-timing-2#dfn-timing-allow-check">timing
-          allow check</a> algorithm when timing information involving the
-          previous navigation is accessed. [[RESOURCE-TIMING-2]]
+          the <a>same-origin check</a> algorithm when timing information
+          involving the previous navigation is accessed. [[RESOURCE-TIMING-2]]
```
Comment (npm1) [inline index.html]: Is the reference to Resource Timing still relevant here?

File: index.html [hunk 2]
```diff
@@ -972,21 +958,13 @@ <h2>
...
-        "resource-timing-2#sec-cross-origin-resources" class=
-        "sec-ref"><span class="secno">4.5</span> <span class=
-        "sec-title">Cross-origin Resources</span></a> [[RESOURCE-TIMING]].
-        Resource providers can explicitly allow all timing information to be
-        collected for a <a>current document</a> by adding the
-        <code>Timing-Allow-Origin</code> HTTP response header, which specifies
-        the domains that are allowed to access the timing information.
+        attributes which include information on the previous document, the
+        <a>same-origin check</a> algorithm is enforced and attributes related
+        to the current document are set to zero.
```
Comment (npm1) [inline index.html]:
```suggestion
        to the previous document are set to zero.
```

It's the ones related to the previous document that are set to zero.

File: index.html [hunk 3]
```diff
@@ -1595,6 +1573,17 @@ <h3>
         </dl>
       </section>
     </section>
+    <section id="same-origin-check">
```
Comment (npm1) [inline index.html]: Let's add this section before the obsolete section.

File: index.html [hunk 4]
```diff
@@ -1595,6 +1573,17 @@ <h3>
...
+      <h3>Same-origin check</h3>
+      <p>
+        When asked to run the <dfn>same-origin check</dfn>, the user agent
+        must run the following: If the previous document, and all <a
```
Comment (npm1) [inline index.html]: Should this have an input or specify 'current document' in a less hand-wavy way? Also, the previous document may not exist, and I don't think we covered all cases where it does not every time we call this, so maybe worth special casing that here too. Given that, we might also want to convert this to an algorithm.

File: index.html (hunk 3 above)
Comment (yoavweiss) [reply index.html]: bumped it up

File: index.html (hunk 4 above)
Comment (yoavweiss) [reply index.html]: There are many underdefined concepts we use here and I don't want to block on them. The current document is used all over the document, previous document is not really defined, there's no link between the current document and the request on which it rode in, and HTTP redirects don't keep track if they were all same-origin or not...

I made it more algorithmy, but a lot of this stuff will have to wait for a larger scale Fetch integration.

File: index.html [hunk 5]
```diff
@@ -336,45 +336,38 @@ <h3>
...
+          <ul>
+            <li>If there is no previous document, or if the <a>same-origin
+              check</a> fails, return a {{DOMHighResTimeStamp}} with a time
+              value equal to zero.</li>    
+            <li>Otherwise, return a <a data-cite=
+              "hr-time-2#dom-domhighrestimestamp">DOMHighResTimeStamp</a> with a
+              time value equal to the time immediately before the user agent starts
+              the <a data-cite=
+              "HTML/browsing-the-web.html#unloading-documents">unload event</a> of
+              the previous document.</li>
```
Comment (npm1) [inline index.html]: nit: missing closing \</ul> (or changed to ol). Seems important to not add indentation to the rest of this section :)

File: index.html [hunk 6]
```diff
@@ -336,45 +336,38 @@ <h3>
...
+        </p>
+        <p data-dfn-for='PerformanceNavigationTiming'>
+          When getting the value of the <dfn>unloadEventEnd</dfn> attribute, run
+          the following steps:
+          <ul>
```
Comment (npm1) [inline index.html]: ditto the 2 above comments

File: index.html [hunk 7]
```diff
@@ -906,6 +902,27 @@ <h3>
         this attack.
       </p>
     </section>
+    <section id="same-origin-check">
+      <h3>Same-origin check</h3>
+      <p>
+        When asked to run the <dfn>same-origin check</dfn>, the user agent
+        MUST run the following steps:
+        <ul>
```
Comment (npm1) [inline index.html]: nit: again ordered list prefered to unordered

File: index.html [hunk 8]
```diff
@@ -906,6 +902,27 @@ <h3>
         this attack.
       </p>
     </section>
+    <section id="same-origin-check">
```
Comment (npm1) [inline index.html]: The Security section is declared non-normative and this is a subsection... maybe add it as a subsection of the "Process" section?

File: index.html [hunk 9]
```diff
@@ -336,45 +336,38 @@ <h3>
...
-      </p>
-      <p data-dfn-for='PerformanceNavigationTiming'>
-        The <dfn>domInteractive</dfn> attribute MUST return a <a
-          data-cite="hr-time-2#dom-domhighrestimestamp">DOMHighResTimeStamp</a> with a
-        time value equal to the time immediately before the user agent sets
-        the <a data-cite="HTML/dom.html#current-document-readiness">current
+        <p data-dfn-for='PerformanceNavigationTiming'>
+          When getting the value of the <dfn>unloadEventStart</dfn> attribute,
+          run the following steps:
+          <ul>
```
[the remark this answers is not in the document]
Comment (yoavweiss) [reply index.html]: indeed

File: index.html (hunk 5 above)
Comment (yoavweiss) [reply index.html]: :+1:

File: index.html (hunk 6 above)
Comment (yoavweiss) [reply index.html]: yup

File: index.html (hunk 8 above)
Comment (yoavweiss) [reply index.html]: good call

**Comment by yoavweiss** [comment]
Related tests are [PRed](https://github.com/web-platform-tests/wpt/pull/17412) as part of a [Chromium CL](https://chromium-review.googlesource.com/c/chromium/src/+/1669355)

**Pull request merged**